webnn/binary.html

```diff
@@ -0,0 +1,1190 @@
+﻿<!doctype html>
+<meta charset=utf-8>
+<title>test element-wise binary operations</title>
+<link rel="help" href="https://webmachinelearning.github.io/webnn/#api-mlgraphbuilder-binary">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="https://webmachinelearning.github.io/webnn-polyfill/dist/webnn-polyfill.js"></script>
+<div id=log></div>
+<script type="module">
+  'use strict';
+  import sizeOfShape from './resources/common.js';
+  import * as baseline from './resources/baseline.js'
+  let builder;
+  setup(() => {
+    const context = navigator.ml.createContext();
+    builder = new MLGraphBuilder(context);
+  })
+
+  test(() => {
+    const dimensions = [3, 4, 5];
+    const inputData0 = [
+      0.08939514,  -1.5887482,  0.8545348,   0.20523034,  -0.41728342,
+      1.01752,     0.19677015,  0.5398451,   0.56893295,  1.2511084,
+      2.0092728,   1.0606714,   0.4893267,   0.09536829,  -2.3467007,
+      2.4527607,   0.61307395,  -1.0799897,  -0.15071101, -0.48422927,
+      -0.20479254, 0.32798728,  -0.37435308, -1.7116562,  1.6952512,
+      -0.7479369,  -0.09019202, 0.14343949,  1.6754607,   1.6427531,
+      0.9470988,   0.20872667,  -1.9530525,  -0.21783416, 0.0309498,
+      0.3008434,   1.1686599,   1.4920886,   0.06633294,  0.6674667,
+      0.60627925,  0.04302086,  -0.03482966, -0.7343786,  -0.76851964,
+      0.9446942,   -0.35489243, 0.44452578,  0.00648887,  -0.55656946,
+      -0.735903,   0.22050636,  -0.5008282,  -1.3132697,  1.6642882,
+      -0.48397836, 0.20099205,  -0.28786168, 1.3315053,   -0.41619393,
+    ];
+    const inputData1 = [
+      -0.5781865,  -0.49248728, -0.2162451,  -0.13176449, -0.52118045,
+      1.9125274,   0.6508799,   0.71873736,  -2.3154447,  0.8080079,
+      0.3022368,   0.21394566,  -0.6511544,  0.20001237,  -0.08041809,
+      1.1127822,   -1.521739,   0.7249548,   -0.91961324, -0.83175105,
+      -1.4569077,  -0.5417681,  -1.6476909,  0.1223801,   2.220618,
+      -0.14914903, 0.7790501,   -0.18711103, -0.9941537,  -1.828552,
+      -1.36035,    0.5727087,   2.5213664,   -0.3267195,  0.8431539,
+      0.12337407,  1.0018097,   -0.23469485, -0.4530751,  0.09238022,
+      0.7888511,   0.11107288,  0.48171726,  0.34308678,  -0.90550417,
+      0.203841,    0.02521433,  -1.7966009,  -1.4287543,  0.3222213,
+      1.0590587,   -1.7948701,  -1.7195907,  -0.9120889,  -0.9391962,
+      -0.2566791,  -0.5464537,  1.4351872,   0.5705938,   -0.30327085,
+    ];
+
+    // get baseline
+    const expected = baseline.binary(
```

Comment:
as I alluded in https://github.com/webmachinelearning/webnn-polyfill/issues/144#issuecomment-985326505, I wonder if it would not be better to provided hardcoded expected values rather than have them be calculated each time by the device under test (e.g. to avoid the case where the CPU or the JS engine itself are faulty). I'm not sure how likely that would be, so maybe this is not a high-priority concern.